File: Srims.WebSite/javascript/srims/Patents/PatentLevel.js

if (!Srims.patents)
    Ext.namespace("Srims.patents");

Ext.namespace('Srims.patents.PatentLevel');

Srims.patents.PatentLevel.Unknown = 'Unknown';
Srims.patents.PatentLevel.TheFirstResponsibleUnion = 'TheFirstResponsibleUnion';
Srims.patents.PatentLevel.Join = 'Join';

Srims.patents.PatentLevelRender = function(value, metadata) {
    switch (value) {
        case 'TheFirstResponsibleUnion':
            return '第一责任单位';
        case 'Join':
            return '参加';
        default:
            return '未识别';
    }
}


Srims.patents.PatentLevelStore = [['TheFirstResponsibleUnion', '第一责任单位'], ['Join', '参加'], ['Unknown', '未识别']];

File: Srims.Server.Business/Patents/PatentLevel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Srims.Server.Business.Patents
{
    /// <summary>
    /// 专利级别
    /// </summary>
    public enum PatentLevel
    {
        /// <summary>
        /// 未识别
        /// </summary>
        Unknown = 0,
        /// <summary>
        /// 第一责任单位
        /// </summary>
        TheFirstResponsibleUnion = 1,
        /// <summary>
        /// 参加
        /// </summary>
        Join = 2
    }
}
